traffic_ops/app/lib/API/Configs/ApacheTrafficServer.pm

```diff
@@ -1424,6 +1426,94 @@ sub drop_qstring_dot_config {
 	return $text;
 }
 
+sub logging_dot_yaml {
+	my $self        = shift;
+	my $profile_obj = shift;
+
+	my $data = $self->profile_param_data( $profile_obj->id, "logging.yaml" );
+
+	# This is an YAML file, so we need to massage the header a bit for YAML commenting.
+	my $text = "# " . $self->header_comment( $profile_obj->name );
+	$text =~ s/# //;
+	$text =~ s/\n//;
+
+	my $max_log_objects = 10;
+
+	# Add formats and filters separately to the top of the file
+	$text .= "\nformats: \n";
+	for ( my $i = 0; $i < $max_log_objects; $i = $i + 1) {
+		my $log_format_field = "LogFormat";
+		if ( $i > 0 ) {
+			$log_format_field = $log_format_field . "$i";
+		}
+		my $log_format_name = $data->{$log_format_field . ".Name"} || "";
+		if ( length($log_format_name) > 0 ) {
+			my $format = $data->{$log_format_field . ".Format"};
+			$text .= " - name: " . $log_format_name ." \n";
+			$text .= "   format: '" . $format . "'\n"
+		}
+	}
+
+	$text .= "filters:\n";
+	for ( my $i = 0; $i < $max_log_objects; $i = $i + 1) {
+		my $log_filter_field = "LogFilter";
+		if ( $i > 0 ) {
+			$log_filter_field = $log_filter_field . "$i";
+		}
+		my $log_filter_name = $data->{$log_filter_field . ".Name"} || "";
+		if ( length($log_filter_name) > 0) {
+			my $filter = $data->{$log_filter_field . ".Filter"};
+			my $log_filter_type = $data->{$log_filter_field . ".Type"} || "accept";
+			$text .= "- name: " . $log_filter_name ."\n";
+			$text .= "  action: " . $log_filter_type . "\n";
+			$text .= "  condition: " . $filter ."\n";
+		}
+	}
+
+	for ( my $i = 0; $i < $max_log_objects; $i = $i + 1 ) {
+		my $log_object_field = "LogObject";
+		if ( $i > 0 ) {
+			$log_object_field = $log_object_field . "$i";
+		}
+
+		my $log_object_filename = $data->{$log_object_field . ".Filename"} || "";
+		if ( length($log_object_filename) > 0 ) {
+			my $log_object_type                 = $data->{$log_object_field . ".Type"}               || "ascii";
+			my $log_object_format               = $data->{$log_object_field . ".Format"}             || "";
+			my $log_object_rolling_enabled      = $data->{$log_object_field . ".RollingEnabled"}     || "";
+			my $log_object_rolling_interval_sec = $data->{$log_object_field . ".RollingIntervalSec"} || "";
+			my $log_object_rolling_offset_hr    = $data->{$log_object_field . ".RollingOffsetHr"}    || "";
+			my $log_object_rolling_size_mb      = $data->{$log_object_field . ".RollingSizeMb"}      || "";
+			my $log_object_filters              = $data->{$log_object_field . ".Filters"}            || "";
```

Comment:
Is it possible for `$log_object_filters` to have a return carriage in it? Cause that would cause problems below.

Reply:
tested now adding \n's and \r's to a filters object and they get written exactly as is into the yaml file and not interpreted as newlines. So I think this is ok as is

Reply:
added stripping of all types of newline, \v repeating, that appears to work